src/crypto/hsm/nxp/PersistentStorageOperationalKeystoreHSM.cpp

```diff
@@ -0,0 +1,288 @@
+    }
+
+    mPendingKeypair = Platform::New<Crypto::P256KeypairHSM>();
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_NO_MEMORY);
+
+    // Key id is created as slotid + start offset of ops key id
+    mPendingKeypair->SetKeyId(FABRIC_SE05X_KEYID_START + slotId);
+
+    err = mPendingKeypair->Initialize();
+    VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_NO_MEMORY);
+
+    mPendingFabricIndex = fabricIndex;
+
+    keyidFabIdMapping[slotId].isPending = 1;
+
+    size_t csrLength = outCertificateSigningRequest.size();
+    err              = mPendingKeypair->NewCertificateSigningRequest(outCertificateSigningRequest.data(), csrLength);
+    if (err != CHIP_NO_ERROR)
+    {
+        ResetPendingKey();
+        if (mPendingKeypair != nullptr)
+        {
+            ResetPendingSlot(mPendingKeypair->GetKeyId() - FABRIC_SE05X_KEYID_START);
+        }
+        return err;
+    }
+    outCertificateSigningRequest.reduce_size(csrLength);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::ActivateOpKeypairForFabric(FabricIndex fabricIndex,
+                                                                               const Crypto::P256PublicKey & nocPublicKey)
+{
+    ChipLogProgress(Crypto, "SE05x: ActivateOpKeypair for Fabric %02x", fabricIndex);
+
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex) && (fabricIndex == mPendingFabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    // Validate public key being activated matches last generated pending keypair
+    VerifyOrReturnError(mPendingKeypair->Pubkey().Matches(nocPublicKey), CHIP_ERROR_INVALID_PUBLIC_KEY);
+
+    mIsPendingKeypairActive = true;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::CommitOpKeypairForFabric(FabricIndex fabricIndex)
+{
+    uint8_t i = 0;
+
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex) && (fabricIndex == mPendingFabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(mIsPendingKeypairActive == true, CHIP_ERROR_INCORRECT_STATE);
+
+    uint32_t slotId = mPendingKeypair->GetKeyId() - FABRIC_SE05X_KEYID_START;
+
+    VerifyOrReturnError(slotId < MAX_KEYID_SLOTS_FOR_FABRICS, CHIP_ERROR_NO_MEMORY);
+
+    ChipLogProgress(Crypto, "SE05x: CommitOpKeypair for Fabric %02x", fabricIndex);
+
+    while (i < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        if (keyidFabIdMapping[i].fabricIndex == fabricIndex)
+        {
+            // Delete the previous keyPair associated with the fabric
+            keyidFabIdMapping[i].isPending = 0;
+            Platform::Delete<Crypto::P256KeypairHSM>(keyidFabIdMapping[i].pkeyPair);
+            keyidFabIdMapping[i].pkeyPair    = NULL;
+            keyidFabIdMapping[i].keyId       = 0;
+            keyidFabIdMapping[i].fabricIndex = 0;
```

Comment:
Shouldn't this use `kUndefinedFabricIndex`?

Reply:
updated.